barretenberg/cpp/src/barretenberg/vm/avm/trace/gadgets/merkle_tree.hpp

```diff
@@ -22,8 +22,14 @@ class AvmMerkleTreeTraceBuilder {
     };
 
     AvmMerkleTreeTraceBuilder() = default;
+    AvmMerkleTreeTraceBuilder(TreeSnapshots& tree_snapshots)
```

Comment:
I think modern cpp practice encourage to std::move in a constructor. @fcarreiro WDYT?

Reply:
If it's a reference it doesn't matter (you can't move it). However, since you are storing a reference you have to be VERY careful about the lifetime (the passed in object should live longer than the whole AvmMerkleTreeTraceBuilder).